rflx/model.py

```diff
@@ -398,9 +345,35 @@ def field_condition(self, field: Field) -> Expr:
     def field_size(self, field: Field) -> Expr:
         if field == FINAL:
             return Number(0)
+
         if field not in self.fields:
             raise ValueError(f'field "{field.name}" not found')
-        return self.types[field].size
+
+        field_type = self.types[field]
+        if isinstance(field_type, Scalar):
+            return field_type.size
+
+        raise NotImplementedError
+
+    def prefixed(self, prefix: str) -> "AbstractMessage":
+        def prefixed_expression(expression: Expr) -> Expr:
+            return expression.simplified(
+                {v: v.__class__(f"{prefix}{v.name}") for v in expression.variables()}
+            )
+
+        structure = []
+
+        for l in self.structure:
+            source = Field(f"{prefix}{l.source.name}") if l.source != INITIAL else INITIAL
+            target = Field(f"{prefix}{l.target.name}") if l.target != FINAL else FINAL
+            condition = prefixed_expression(l.condition)
+            length = prefixed_expression(l.length) if l.length != UNDEFINED else UNDEFINED
+            first = prefixed_expression(l.first) if l.first != UNDEFINED else UNDEFINED
```

Comment:
Would it make sense to remove this check and implement `simplified` for `UNDEFINED` in a way that it always returns `UNDEFINED`?

Reply:
Good hint. Actually, `simplified` is already behaving like that (if no explicit substitution for `UNDEFINED` is given as argument) and thus this check is not needed here.

Reply:
Fixed by ee83374.